x509/src/certificate.rs

```diff
@@ -121,27 +192,45 @@ impl<'a> ::der::Sequence<'a> for TBSCertificate<'a> {
     where
         F: FnOnce(&[&dyn der::Encodable]) -> ::der::Result<T>,
     {
+        let version_value: u8 = self.version.try_into()?;
+
+        let issuer_unique_id = if Version::V2 <= self.version {
+            &self.issuer_unique_id
+        } else {
+            &None
+        };
+        let subject_unique_id = if Version::V2 <= self.version {
+            &self.subject_unique_id
+        } else {
+            &None
+        };
+        let extensions = if Version::V3 <= self.version {
+            self.extensions.as_ref().map(|exts| ContextSpecific {
+                tag_number: EXTENSIONS_TAG,
+                tag_mode: TagMode::Explicit,
+                value: exts.clone(),
+            })
+        } else {
+            None
+        };
+
         #[allow(unused_imports)]
         use core::convert::TryFrom;
         f(&[
             &ContextSpecific {
                 tag_number: VERSION_TAG,
                 tag_mode: TagMode::Explicit,
-                value: self.version,
+                value: version_value,
```

Comment:
Seems like just `self.version` should work here, especially if you make it `Copy`

Reply:
Addressed here (initially I though you meant replace that production with self.version but landed as shown): https://github.com/carl-wallace/formats/commit/c226d37a77913bad7a0a674f679b7b85a4ecb280